anydex/test/wallet/stellar/test_stellar_wallet.py

```diff
@@ -0,0 +1,308 @@
+                                          "97YgM+Be8yponPH0YjMjeYphewI",
+                     fee=100,
+                     is_pending=False,
+                     succeeded=True,
+                     sequence_number=112092925529161789,
+                     min_time_bound=datetime.fromisoformat('1970-01-01T00:00:00'))
+
+    def test_get_identifier(self):
+        self.assertEqual(self.expected_identifier(), self.wallet.get_identifier())
+
+    @abc.abstractmethod
+    def new_wallet(self):
+        """
+        Create a new wallet
+        :return: the new wallet
+        """
+
+    @abc.abstractmethod
+    def expected_identifier(self):
+        return
+
+    def create_wallet(self):
+        """
+        Create the wallet
+        """
+        self.wallet.get_sequence_number = lambda: 100
+        self.wallet.create_wallet()
+
+    def test_create_wallet(self):
+        """
+        Test for wallet creation
+        """
+
+        self.create_wallet()
+        addr = self.wallet.stellar_db.session.query(Secret.address).first()[0]
+        self.assertIsNotNone(self.wallet.keypair)
+        self.assertTrue(self.wallet.created)
+        self.assertEqual(addr, self.wallet.get_address())
+
+    def test_init_wallet_created(self):
+        """
+        Test for the wallet constructor when the wallet is already created
+        """
+        self.create_wallet()
+        StellarWallet.get_sequence_number = lambda *_: 100
+
+        wallet = self.new_wallet()
+        self.assertEqual(wallet.created, True)
+
+    def test_create_wallet_already_created(self):
+        """
+        Test for wallet creation where we tried to create a wallet that is  already created
+        """
+        self.create_wallet()
+        future = self.wallet.create_wallet()
+
+        self.assertIsInstance(future.exception(), RuntimeError)
+
+    @abc.abstractmethod
+    def test_get_name(self):
+        return
+
+    async def test_get_balance_not_created(self):
+        """
+        Test for getting balance when the wallet has not yet been created
+        """
+        balance = {
+            'available': 0,
+            'pending': 0,
+            'currency': self.expected_identifier(),
+            'precision': 7
+        }
+        self.assertEquals(balance, await self.wallet.get_balance())
+
+    async def test_get_balance(self):
+        """
+        Test for getting the balance when the wallet has been created
+        """
+        self.wallet.provider = MockObject()
+        self.create_wallet()
+        self.wallet.provider.get_balance = lambda *_, **x: 100
+        balance = {
+            'available': 100 * 1e7,  # balance form api is not in the lowest denomination
+            'pending': 0,
+            'currency': self.expected_identifier(),
+            'precision': 7
+        }
+        self.assertEquals(balance, await self.wallet.get_balance())
+
+    def test_get_sequence_number_db(self):
```

Comment:
Test fails with `expected: 1, actual: 100` for concrete wallet class, not testnet class.